linkerd/app/gateway/src/make.rs

```diff
@@ -64,46 +87,65 @@ where
             }
         };
 
-        let suffixes = self.suffixes.clone();
-        let mut outbound = self.outbound.clone();
-        Box::new(
-            // First, resolve the original name. This determines both the
-            // canonical name as well as an IP that can be used as the outbound
-            // original dst.
-            self.resolve
-                .call(orig_dst.name().clone())
-                .map_err(Into::into)
-                .and_then(move |(name, dst_ip)| {
-                    tracing::debug!(%name, %dst_ip, "Resolved");
-                    if !suffixes.iter().any(|s| s.contains(&name)) {
-                        tracing::debug!(%name, ?suffixes, "No matches");
-                        return future::Either::A(future::ok(Gateway::BadDomain(name)));
-                    }
+        match self {
+            Self::NoSuffixes => {
+                return Box::new(future::ok(Gateway::BadDomain(orig_dst.name().clone())));
+            }
+            Self::NoIdentity => {
+                return Box::new(future::ok(Gateway::NoIdentity));
+            }
+            Self::Enabled {
+                ref mut resolve,
+                outbound,
+                suffixes,
+                local_identity,
+            } => {
+                let mut outbound = outbound.clone();
+                let local_identity = local_identity.clone();
+                let suffixes = suffixes.clone();
+                Box::new(
+                    // First, resolve the original name. This determines both the
+                    // canonical name as well as an IP that can be used as the outbound
+                    // original dst.
+                    resolve
+                        .call(orig_dst.name().clone())
```

Comment:
should we stick a trace span on this for the original destinaton name, or is that added somewhere else in the stack?

Reply:
Yep, we get that on the stack ;)

https://github.com/linkerd/linkerd2-proxy/blob/9a9fa33718dc3c612ac8108420d55d34e83e3816/linkerd/app/outbound/src/lib.rs#L118